File: Napakalaki/src/NapakalakiGame/Dice.java
/**
 * Dice.java
 */

package NapakalakiGame;

import java.util.Random;

/**
 * Representa el dado. 
 * Singleton.
 */
public class Dice {
    
    private static Dice instance = null;
    private final Random random;
    
    private Dice() {
        random = new Random();
    }
    
    public static Dice getInstance() {
        if (instance == null) {
            instance = new Dice();
        }
         
        return instance;    
    }
    
    public int nextNumber() {
        return 1 + random.nextInt(6);
    }
    
}

File: Napakalaki/lib/dice.rb
# encoding: utf-8

require 'singleton'

module NapakalakiGame
  
  # class Dice: representa el dado
  class Dice
    include Singleton  # la clase es <<singleton>>

    # devuelve un entero pseudo-aleatorio del intervalo [1,6]
    def nextNumber
      rand(1..6)                 # ruby >= 1.9
      # (rand() * 6 + 1).to_i    # ruby < 1.9
    end

  end # Dice

end # NapakalakiGame
